milli/src/fields_ids_map.rs

```diff
@@ -65,7 +65,7 @@ impl FieldsIdsMap {
     }
 
     /// Iterate over the ids in the order of the ids.
-    pub fn ids<'a>(&'a self) -> impl Iterator<Item = FieldId> + 'a {
+    pub fn ids(&'_ self) -> impl Iterator<Item = FieldId> + '_ {
```

Comment:
Is putting an anonymous lifetime necessary here?
```suggestion
    pub fn ids(&self) -> impl Iterator<Item = FieldId> {
```

Reply:
This one is needed:

```
hidden type for `impl Iterator<Item = u16>` captures lifetime that does not appear in boundsrustc[E0700](https://doc.rust-lang.org/error-index.html#E0700)
fields_ids_map.rs(68, 16): hidden type `std::iter::Copied<std::collections::btree_map::Keys<'_, u16, std::string::String>>` captures the anonymous lifetime defined here
fields_ids_map.rs(68, 55): to declare that the `impl Trait` captures `'_`, you can add an explicit `'_` lifetime bound: ` + '_`
```